src/pudl_archiver/zenodo/api_client.py

```diff
@@ -178,23 +165,39 @@ async def create_deposition(self, data_source_id: str) -> Deposition:
         metadata = DepositionMetadata.from_data_source(data_source_id)
         return await self.depositor.create_deposition(metadata)
 
-    async def add_files(self, resources: dict[str, ResourceInfo]):
-        """Add all downloaded files to zenodo deposition.
+    async def run(self):
+        """Run the entire deposition update process.
 
-        The new Zenodo deposition (self.new_deposition) contains all of the files
-        from the previous version. This function will loop through all downloaded
-        files and compare them by name and checksum to the remote files already
-        contained in the deposition. It will then upload new files, update files that
-        already exist, and delete any files that are no longer contained in the new
-        deposition.
-
-        Args:
-            resources: Dictionary mapping filenames to ResourceInfo objects,
-            which contain local path to resource and working partitions.
+        1. Create pending deposition version to stage changes.
+        2. Download resources.
+        3. Update pending deposition version.
+        4. If there were updates, publish.
+        5. Update the dataset settings if this was a new deposition.
 
         Returns:
-            Updated Deposition object.
+            Updated Deposition object - if published, then return that; else
+            return the pending deposition version. Distinguishable by `state`
+            field being 'done' and 'unsubmitted', respectively.
         """
+        await self._initialize()
```

Comment:
I also wonder if we could pass around `new_deposition` and `depostion_files` as well as the changes. It feels like with this new flow we can rip out almost all of the state from the deposition orchestrator